### PR TITLE
Refactor Arrow schema conversion

### PR DESCRIPTION
**Pull request by Fokko**
We wrapped a schema in a schema.

### CODE REVIEW

File: pyiceberg/io/pyarrow.py [hunk 1]
```diff
@@ -435,13 +435,18 @@ def delete(self, location: Union[str, InputFile, OutputFile]) -> None:
             raise  # pragma: no cover - If some other kind of OSError, raise the raw error
 
 
-def schema_to_pyarrow(schema: Union[Schema, IcebergType]) -> pa.schema:
-    return visit(schema, _ConvertToArrowSchema())
+def schema_to_pyarrow(schema: Union[Schema, IcebergType], metadata: Dict[bytes, bytes] = EMPTY_DICT) -> pa.schema:
+    return visit(schema, _ConvertToArrowSchema(metadata))
```
Comment (bitsondatadev) [inline pyiceberg/io/pyarrow.py:439]: What is the `visit()` behavior with an empty dict?

**Review by bitsondatadev — COMMENTED** [review]
Nits

File: tests/io/test_pyarrow.py [hunk 2]
```diff
@@ -708,15 +709,17 @@ def _write_table_to_file(filepath: str, schema: pa.Schema, table: pa.Table) -> s
 
 @pytest.fixture
 def file_int(schema_int: Schema, tmpdir: str) -> str:
-    pyarrow_schema = pa.schema(schema_to_pyarrow(schema_int), metadata={"iceberg.schema": schema_int.model_dump_json()})
+    pyarrow_schema = schema_to_pyarrow(schema_int, metadata={ICEBERG_SCHEMA: bytes(schema_int.model_dump_json(), 'utf-8')})
```
[the remark this answers is not in the document]
Comment (Fokko) [reply tests/io/test_pyarrow.py:712]: Sounds good, I've introduced a utf8 constant 👍

File: pyiceberg/io/pyarrow.py (hunk 1 above)
Comment (Fokko) [reply pyiceberg/io/pyarrow.py:439]: Sometime we use the visitor to convert types, then we don't need to set any metadata so then a default with an empty dict makes things easier and less verbose.

**Review by bitsondatadev — APPROVED** [review]
LGTM!

**Review by amogh-jahagirdar — APPROVED** [review]

**Comment by Fokko** [comment]
Thanks @bitsondatadev and @amogh-jahagirdar for the review 🥳

**Pull request merged**